tests/azure_iot_provisioning_client_ut.c

```diff
@@ -637,6 +665,64 @@ static void testAzureIoTProvisioningClient_Register_QueryFailure( void ** ppvSta
 }
 /*-----------------------------------------------------------*/
 
+static void testAzureIoTProvisioningClient_Register_InvalidResponseFailure( void ** ppvState )
+{
+    AzureIoTProvisioningClient_t xTestProvisioningClient;
+    AzureIoTMQTTPublishInfo_t xPublishInfo;
+
+    ( void ) ppvState;
+
+    prvSetupTestProvisioningClient( &xTestProvisioningClient );
+
+    /* Connect */
+    prvRegistrationConnectStep( &xTestProvisioningClient );
+
+    /* Subscribe */
+    prvRegistrationSubscribeStep( &xTestProvisioningClient );
+
+    /* AckSubscribe */
+    prvRegistrationAckSubscribeStep( &xTestProvisioningClient );
+
+    /* Publish Registration Request */
+    prvRegistrationPublishStep( &xTestProvisioningClient );
+
+    /* Registration response */
+    prvGenerateGoodResponse( &xPublishInfo, 1 );
+    assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient, 0 ),
```

Comment:
nit: can we have a macro for `0` in the test file for no wait? ex
```suggestion
    assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient, provtestNO_WAIT_TIMEOUT ),
```
easier to read and remember what that param is doing

Reply:
I think better will be to have marco from the provisioning header.